### ISSUE TITLE
Creation an object with `ignore_on_error` flag set causes an infinite recursion

### ISSUE THREAD

**Issue by yhabteab** [Icinga/icinga2#8824]
## Describe the bug

When an error is encountered while creating an object, for example due to forgetting the ``check_command`` attribute for a host object and having ``ignore_on_error`` flag set, it unfortunately causes an endless recursion in [ConfigItem::CommitNewItems()](https://github.com/Icinga/icinga2/blob/275ba6d1df3219158818f77109e131f36c742abe/lib/config/configitem.cpp#L386) and finally leads Icinga2 to crash. It is normally the case that if an error occurs in [ConfigItem::Commit()](https://github.com/Icinga/icinga2/blob/275ba6d1df3219158818f77109e131f36c742abe/lib/config/configitem.cpp#L162) when evaluating or validating the config and the ``ignore_on_error`` flag is not set, immediately an error is raised but if this flag is set, a ``nullptr`` is returned instead. Since this returned value is not checked in [CommitNewItems()](https://github.com/Icinga/icinga2/blob/275ba6d1df3219158818f77109e131f36c742abe/lib/config/configitem.cpp#L472) it will always continue even though the object could not be created.

And as ``CommitNewItems()`` calls itself recursively and the config item which could not be created has not been committed and hence is not yet a config object, it will be considered as a new object over and over again. This procedure will be repeated until it most likely ends up hitting a corrupt memory and crashing icinga2 because none of the conditions will break this infinite recursion call unless an exception is raised in the ``WorkQueue``.

However, I'm still not sure where exactly this crash occurs, as it only happens occasionally and the ``nullptr`` value being returned is nowhere direferenced.

**Comment by Yonas-net** [comment]
ping @Al2Klimov!

**Comment by Al2Klimov** [comment]
If `ignore_on_error` crashes API objects, how did regular config objects ever work?

**Comment by Yonas-net** [comment]
> If `ignore_on_error` crashes API objects, how did regular config objects ever work?

As I already said, the problem appears only after this fix.
> You can only reproduce the error with this fix #8819.

**Comment by Yonas-net** [comment]
Indeed, I don't think anyone sets the ``ignore_on_error`` flag even on Manuel created objects, but Icinga2 will still not continue anyway, even if the flag is set, when error occurs in the configs.

```powershell
[2021-06-21 12:33:39 +0200] notice/ConfigObject: Ignoring config object 'icinga2-host' of type 'Host' due to errors: Error: Validation failed for object 'icinga2-host' of type 'Host'; Attribute 'check_command': Attribute must not be empty.
Location: in /Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf: 18:1-18:42
/Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf(16):  */
/Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf(17): 
/Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf(18): object Host "icinga2-host" ignore_on_error {
                                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf(19):     address = "localhost"
/Users/yhabteab/Workspace/icinga2/prefix/etc/icinga2/conf.d/hosts.conf(20):     vars.os = "Linux"
[2021-06-21 12:33:39 +0200] notice/cli: Worker process couldn't load its config
```
